common/spec/dependabot/file_fetchers/base_spec.rb

```diff
@@ -1313,4 +1319,82 @@ def fetch_files
       end
     end
   end
+
+  context "with repo_contents_path" do
+    let(:repo_contents_path) { Dir.mktmpdir }
+    # prepare a checkout to spoof `git clone`
+    let(:contents) { fixture("ruby", "gemfiles", "Gemfile") }
+    let(:mock_clone) do
+      `git init #{repo_contents_path}`
+      path = File.join(repo_contents_path, "requirements.txt")
+      File.write(path, contents)
+      `git -C #{repo_contents_path} add .`
+      `git -C #{repo_contents_path} commit -m'fake clone'`
+      repo_contents_path
+    end
```

Comment:
Since the `"file not found"` context and the `"symlink"` context define their own `let(:mock_clone`) block, I think that means that this block only applies to the `"with a GitHub source` context. How would you feel about moving this let block into the `"with a GitHub source` context to make it easier to see that it is specific to that context?

Reply:
I didn't like the idea of the `before` block in this context referencing `mock_clone` that only existed in child contexts.

Per the other discussion thread though, `mock_clone` is attempted cleverness that should likely be reworked.

Reply:
I've updated to DRY this: 
* `mock_clone` still exists here, in the context it's used
* `fill_repo` is introduced to do the custom bits between each context

`fill_repo` might be a little too close to the flagged pattern of abusing `let` for side effects though, what do you think?

Reply:
I dig those updates, @thepwagner. 👍 

The changes overall look good to me, but I don't feel like I know enough about the big picture yet to offer a meaningful code review approval, so I'd like to give @jurre a chance to review it.